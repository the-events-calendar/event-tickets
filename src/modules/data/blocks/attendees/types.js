/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

export const SET_ATTENDEES_INITIAL_STATE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_INITIAL_STATE`;

export const SET_ATTENDEES_TITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_TITLE`;

export const SET_ATTENDEES_DISPLAY_TITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_DISPLAY_TITLE`;

export const SET_ATTENDEES_DISPLAY_SUBTITLE = `${ PREFIX_TICKETS_STORE }/SET_ATTENDEES_DISPLAY_SUBTITLE`;
