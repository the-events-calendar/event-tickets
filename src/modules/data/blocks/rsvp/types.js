/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

//
// ─── RSVP TYPES ─────────────────────────────────────────────────────────────────
//

export const SET_RSVP_ID = `${ PREFIX_TICKETS_STORE }/SET_RSVP_ID`;
export const SET_RSVP_SETTINGS_OPEN = `${ PREFIX_TICKETS_STORE }/SET_RSVP_SETTINGS_OPEN`;
export const SET_RSVP_HAS_CHANGES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_CHANGES`;
export const SET_RSVP_IS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_LOADING`;
export const SET_RSVP_IS_SETTINGS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_SETTINGS_LOADING`;
export const SET_RSVP_IS_MODAL_OPEN = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_MODAL_OPEN`;
export const SET_RSVP_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_GOING_COUNT`;
export const SET_RSVP_NOT_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_COUNT`;
export const SET_RSVP_HAS_ATTENDEE_INFO_FIELDS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_ATTENDEE_INFO_FIELDS`;
export const SET_RSVP_HAS_DURATION_ERROR = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_DURATION_ERROR`;

export const SET_RSVP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DETAILS`;
export const SET_RSVP_TEMP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DETAILS`;
export const SET_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HEADER_IMAGE`;

//
// ─── RSVP DETAILS TYPES ─────────────────────────────────────────────────────────
//

export const SET_RSVP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TITLE`;
export const SET_RSVP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DESCRIPTION`;
export const SET_RSVP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_CAPACITY`;
export const SET_RSVP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_RESPONSES`;
export const SET_RSVP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE`;
export const SET_RSVP_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE_INPUT`;
export const SET_RSVP_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE_MOMENT`;
export const SET_RSVP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE`;
export const SET_RSVP_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE_INPUT`;
export const SET_RSVP_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE_MOMENT`;
export const SET_RSVP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_TIME`;
export const SET_RSVP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_TIME`;
export const SET_RSVP_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_TIME_INPUT`;
export const SET_RSVP_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_TIME_INPUT`;

//
// ─── RSVP TEMP DETAILS TYPES ────────────────────────────────────────────────────
//

export const SET_RSVP_TEMP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_TITLE`;
export const SET_RSVP_TEMP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DESCRIPTION`;
export const SET_RSVP_TEMP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_CAPACITY`;
export const SET_RSVP_TEMP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_NOT_GOING_RESPONSES`;
export const SET_RSVP_TEMP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE`;
export const SET_RSVP_TEMP_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE_INPUT`;
export const SET_RSVP_TEMP_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE_MOMENT`;
export const SET_RSVP_TEMP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE`;
export const SET_RSVP_TEMP_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE_INPUT`;
export const SET_RSVP_TEMP_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE_MOMENT`;
export const SET_RSVP_TEMP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_TIME`;
export const SET_RSVP_TEMP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_TIME`;
export const SET_RSVP_TEMP_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_TIME_INPUT`;
export const SET_RSVP_TEMP_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_TIME_INPUT`;

//
// ─── RSVP THUNK & SAGA TYPES ────────────────────────────────────────────────────
//

export const CREATE_RSVP = `${ PREFIX_TICKETS_STORE }/CREATE_RSVP`;
export const INITIALIZE_RSVP = `${ PREFIX_TICKETS_STORE }/INITIALIZE_RSVP`;
export const DELETE_RSVP = `${ PREFIX_TICKETS_STORE }/DELETE_RSVP`;

export const HANDLE_RSVP_START_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_START_DATE`;
export const HANDLE_RSVP_END_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_END_DATE`;
export const HANDLE_RSVP_START_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_START_TIME`;
export const HANDLE_RSVP_END_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_RSVP_END_TIME`;

export const FETCH_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/FETCH_RSVP_HEADER_IMAGE`;
export const UPDATE_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/UPDATE_RSVP_HEADER_IMAGE`;
export const DELETE_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/DELETE_RSVP_HEADER_IMAGE`;
