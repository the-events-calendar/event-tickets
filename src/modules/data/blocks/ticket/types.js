/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

//
// ─── TICKETS TYPES ──────────────────────────────────────────────────────────────
//

export const SET_TICKETS_INITIAL_STATE = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_INITIAL_STATE`;
export const RESET_TICKETS_BLOCK = `${ PREFIX_TICKETS_STORE }/RESET_TICKETS_BLOCK`;

export const SET_TICKETS_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_HEADER_IMAGE`;
export const SET_TICKETS_IS_SELECTED = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_IS_SELECTED`;
export const SET_TICKETS_IS_SETTINGS_OPEN = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_IS_SETTINGS_OPEN`;
export const SET_TICKETS_IS_SETTINGS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_IS_SETTINGS_LOADING`;
export const SET_TICKETS_PROVIDER = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_PROVIDER`;
export const SET_TICKETS_SHARED_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_SHARED_CAPACITY`;
export const SET_TICKETS_TEMP_SHARED_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_TICKETS_TEMP_SHARED_CAPACITY`;

//
// ─── HEADER IMAGE SAGA TYPES ────────────────────────────────────────────────────
//

export const FETCH_TICKETS_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/FETCH_TICKETS_HEADER_IMAGE`;
export const UPDATE_TICKETS_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/UPDATE_TICKETS_HEADER_IMAGE`;
export const DELETE_TICKETS_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/DELETE_TICKETS_HEADER_IMAGE`;

//
// ─── CHILD TICKET TYPES ─────────────────────────────────────────────────────────
//

export const REGISTER_TICKET_BLOCK = `${ PREFIX_TICKETS_STORE }/REGISTER_TICKET_BLOCK`;
export const REMOVE_TICKET_BLOCK = `${ PREFIX_TICKETS_STORE }/REMOVE_TICKET_BLOCK`;
export const REMOVE_TICKET_BLOCKS = `${ PREFIX_TICKETS_STORE }/REMOVE_TICKET_BLOCKS`;

export const SET_TICKET_TITLE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TITLE`;
export const SET_TICKET_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_TICKET_DESCRIPTION`;
export const SET_TICKET_PRICE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_PRICE`;
export const SET_TICKET_SKU = `${ PREFIX_TICKETS_STORE }/SET_TICKET_SKU`;
export const SET_TICKET_IAC_SETTING = `${ PREFIX_TICKETS_STORE }/SET_TICKET_IAC_SETTING`;
export const SET_TICKET_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_START_DATE`;
export const SET_TICKET_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_START_DATE_INPUT`;
export const SET_TICKET_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_START_DATE_MOMENT`;
export const SET_TICKET_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_END_DATE`;
export const SET_TICKET_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_END_DATE_INPUT`;
export const SET_TICKET_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_END_DATE_MOMENT`;
export const SET_TICKET_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_TICKET_START_TIME`;
export const SET_TICKET_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_TICKET_END_TIME`;
export const SET_TICKET_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_START_TIME_INPUT`;
export const SET_TICKET_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_END_TIME_INPUT`;
export const SET_TICKET_CAPACITY_TYPE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_CAPACITY_TYPE`;
export const SET_TICKET_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_TICKET_CAPACITY`;

export const SET_TICKET_TEMP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_TITLE`;
export const SET_TICKET_TEMP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_DESCRIPTION`;
export const SET_TICKET_TEMP_PRICE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_PRICE`;
export const SET_TICKET_TEMP_SKU = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_SKU`;
export const SET_TICKET_TEMP_IAC_SETTING = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_IAC_SETTING`;
export const SET_TICKET_TEMP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_START_DATE`;
export const SET_TICKET_TEMP_START_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_START_DATE_INPUT`;
export const SET_TICKET_TEMP_START_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_START_DATE_MOMENT`;
export const SET_TICKET_TEMP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_END_DATE`;
export const SET_TICKET_TEMP_END_DATE_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_END_DATE_INPUT`;
export const SET_TICKET_TEMP_END_DATE_MOMENT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_END_DATE_MOMENT`;
export const SET_TICKET_TEMP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_START_TIME`;
export const SET_TICKET_TEMP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_END_TIME`;
export const SET_TICKET_TEMP_START_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_START_TIME_INPUT`;
export const SET_TICKET_TEMP_END_TIME_INPUT = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_END_TIME_INPUT`;
export const SET_TICKET_TEMP_CAPACITY_TYPE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_CAPACITY_TYPE`;
export const SET_TICKET_TEMP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_CAPACITY`;

export const SET_TICKET_SOLD = `${ PREFIX_TICKETS_STORE }/SET_TICKET_SOLD`;
export const SET_TICKET_AVAILABLE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_AVAILABLE`;
export const SET_TICKET_ID = `${ PREFIX_TICKETS_STORE }/SET_TICKET_ID`;
export const SET_TICKET_CURRENCY_SYMBOL = `${ PREFIX_TICKETS_STORE }/SET_TICKET_CURRENCY_SYMBOL`;
export const SET_TICKET_CURRENCY_POSITION = `${ PREFIX_TICKETS_STORE }/SET_TICKET_CURRENCY_POSITION`;
export const SET_TICKET_PROVIDER = `${ PREFIX_TICKETS_STORE }/SET_TICKET_PROVIDER`;
export const SET_TICKET_HAS_ATTENDEE_INFO_FIELDS = `${ PREFIX_TICKETS_STORE }/SET_TICKET_HAS_ATTENDEE_INFO_FIELDS`;
export const SET_TICKET_IS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_TICKET_IS_LOADING`;
export const SET_TICKET_IS_MODAL_OPEN = `${ PREFIX_TICKETS_STORE }/SET_TICKET_IS_MODAL_OPEN`;
export const SET_TICKET_HAS_BEEN_CREATED = `${ PREFIX_TICKETS_STORE }/SET_TICKET_HAS_BEEN_CREATED`;
export const SET_TICKET_HAS_CHANGES = `${ PREFIX_TICKETS_STORE }/SET_TICKET_HAS_CHANGES`;
export const SET_TICKET_HAS_DURATION_ERROR = `${ PREFIX_TICKETS_STORE }/SET_TICKET_HAS_DURATION_ERROR`;
export const SET_TICKET_IS_SELECTED = `${ PREFIX_TICKETS_STORE }/SET_TICKET_IS_SELECTED`;

//
// ─── CHILD TICKET SAGA TYPES ────────────────────────────────────────────────────
//

export const SET_TICKET_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_TICKET_DETAILS`;
export const SET_TICKET_TEMP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_TICKET_TEMP_DETAILS`;

export const HANDLE_TICKET_START_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_TICKET_START_DATE`;
export const HANDLE_TICKET_END_DATE = `${ PREFIX_TICKETS_STORE }/HANDLE_TICKET_END_DATE`;
export const HANDLE_TICKET_START_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_TICKET_START_TIME`;
export const HANDLE_TICKET_END_TIME = `${ PREFIX_TICKETS_STORE }/HANDLE_TICKET_END_TIME`;

export const FETCH_TICKET = `${ PREFIX_TICKETS_STORE }/FETCH_TICKET`;
export const CREATE_NEW_TICKET = `${ PREFIX_TICKETS_STORE }/CREATE_NEW_TICKET`;
export const UPDATE_TICKET = `${ PREFIX_TICKETS_STORE }/UPDATE_TICKET`;
export const DELETE_TICKET = `${ PREFIX_TICKETS_STORE }/DELETE_TICKET`;

export const SET_TICKET_INITIAL_STATE = `${ PREFIX_TICKETS_STORE }/SET_TICKET_INITIAL_STATE`;
