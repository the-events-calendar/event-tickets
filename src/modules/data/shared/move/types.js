/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '../../utils';

//
// ─── MODAL DATA ─────────────────────────────────────────────────────────────────
//

export const SET_MODAL_DATA = `${ PREFIX_TICKETS_STORE }/SET_MODAL_DATA`;
export const RESET_MODAL_DATA = `${ PREFIX_TICKETS_STORE }/RESET_MODAL_DATA`;
export const SUBMIT_MODAL = `${ PREFIX_TICKETS_STORE }/SUBMIT_MODAL`;

//
// ─── MODAL UI STATE ─────────────────────────────────────────────────────────────
//

export const INITIALIZE_MODAL = `${ PREFIX_TICKETS_STORE }/INITIALIZE_MODAL`;
export const SHOW_MODAL = `${ PREFIX_TICKETS_STORE }/SHOW_MODAL`;
export const HIDE_MODAL = `${ PREFIX_TICKETS_STORE }/HIDE_MODAL`;

//
// ─── API ───────────────────────────────────────────────────────────────────────-
//

export const FETCH_POST_TYPES = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES`;
export const FETCH_POST_TYPES_SUCCESS = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES_SUCCESS`;
export const FETCH_POST_TYPES_ERROR = `${ PREFIX_TICKETS_STORE }/FETCH_POST_TYPES_ERROR`;

export const FETCH_POST_CHOICES = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES`;
export const FETCH_POST_CHOICES_SUCCESS = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES_SUCCESS`;
export const FETCH_POST_CHOICES_ERROR = `${ PREFIX_TICKETS_STORE }/FETCH_POST_CHOICES_ERROR`;

export const MOVE_TICKET = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET`;
export const MOVE_TICKET_SUCCESS = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET_SUCCESS`;
export const MOVE_TICKET_ERROR = `${ PREFIX_TICKETS_STORE }/MOVE_TICKET_ERROR`;
