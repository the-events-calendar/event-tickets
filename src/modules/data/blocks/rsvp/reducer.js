/**
 * Internal dependencies
 */
import details, { DEFAULT_STATE as DETAILS_DEFAULT_STATE } from './reducers/details';
import tempDetails from './reducers/temp-details';
import headerImage, { DEFAULT_STATE as HEADER_IMAGE_DEFAULT_STATE } from './reducers/header-image';
import { types } from './index';

export const DEFAULT_STATE = {
	id: 0,
	created: false,
	settingsOpen: false,
	hasChanges: false,
	isAddEditOpen: false,
	isLoading: false,
	isSettingsLoading: false,
	isModalOpen: false,
	goingCount: 0,
	notGoingCount: 0,
	hasAttendeeInfoFields: false,
	details: DETAILS_DEFAULT_STATE,
	tempDetails: DETAILS_DEFAULT_STATE,
	headerImage: HEADER_IMAGE_DEFAULT_STATE,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.CREATE_RSVP:
			return {
				...state,
				created: true,
			};
		case types.DELETE_RSVP:
			return DEFAULT_STATE;
		case types.SET_RSVP_ID:
			return {
				...state,
				id: action.payload.id,
			};
		case types.SET_RSVP_SETTINGS_OPEN:
			return {
				...state,
				settingsOpen: action.payload.settingsOpen,
			};
		case types.SET_RSVP_IS_ADD_EDIT_OPEN:
			return {
				...state,
				isAddEditOpen: action.payload.isAddEditOpen,
			};
		case types.SET_RSVP_HAS_CHANGES:
			return {
				...state,
				hasChanges: action.payload.hasChanges,
			};
		case types.SET_RSVP_IS_LOADING:
			return {
				...state,
				isLoading: action.payload.isLoading,
			};
		case types.SET_RSVP_IS_SETTINGS_LOADING:
			return {
				...state,
				isSettingsLoading: action.payload.isSettingsLoading,
			};
		case types.SET_RSVP_IS_MODAL_OPEN:
			return {
				...state,
				isModalOpen: action.payload.isModalOpen,
			};
		case types.SET_RSVP_GOING_COUNT:
			return {
				...state,
				goingCount: action.payload.goingCount,
			};
		case types.SET_RSVP_NOT_GOING_COUNT:
			return {
				...state,
				notGoingCount: action.payload.notGoingCount,
			};
		case types.SET_RSVP_HAS_ATTENDEE_INFO_FIELDS:
			return {
				...state,
				hasAttendeeInfoFields: action.payload.hasAttendeeInfoFields,
			};
		case types.SET_RSVP_HAS_DURATION_ERROR:
			return {
				...state,
				hasDurationError: action.payload.hasDurationError,
			};
		case types.SET_RSVP_TITLE:
		case types.SET_RSVP_DESCRIPTION:
		case types.SET_RSVP_CAPACITY:
		case types.SET_RSVP_NOT_GOING_RESPONSES:
		case types.SET_RSVP_START_DATE:
		case types.SET_RSVP_START_DATE_INPUT:
		case types.SET_RSVP_START_DATE_MOMENT:
		case types.SET_RSVP_END_DATE:
		case types.SET_RSVP_END_DATE_INPUT:
		case types.SET_RSVP_END_DATE_MOMENT:
		case types.SET_RSVP_START_TIME:
		case types.SET_RSVP_END_TIME:
		case types.SET_RSVP_START_TIME_INPUT:
		case types.SET_RSVP_END_TIME_INPUT:
			return {
				...state,
				details: details( state.details, action ),
			};
		case types.SET_RSVP_TEMP_TITLE:
		case types.SET_RSVP_TEMP_DESCRIPTION:
		case types.SET_RSVP_TEMP_CAPACITY:
		case types.SET_RSVP_TEMP_NOT_GOING_RESPONSES:
		case types.SET_RSVP_TEMP_START_DATE:
		case types.SET_RSVP_TEMP_START_DATE_INPUT:
		case types.SET_RSVP_TEMP_START_DATE_MOMENT:
		case types.SET_RSVP_TEMP_END_DATE:
		case types.SET_RSVP_TEMP_END_DATE_INPUT:
		case types.SET_RSVP_TEMP_END_DATE_MOMENT:
		case types.SET_RSVP_TEMP_START_TIME:
		case types.SET_RSVP_TEMP_END_TIME:
		case types.SET_RSVP_TEMP_START_TIME_INPUT:
		case types.SET_RSVP_TEMP_END_TIME_INPUT:
			return {
				...state,
				tempDetails: tempDetails( state.tempDetails, action ),
			};
		case types.SET_RSVP_HEADER_IMAGE:
			return {
				...state,
				headerImage: headerImage( state.headerImage, action ),
			};
		default:
			return state;
	}
};
