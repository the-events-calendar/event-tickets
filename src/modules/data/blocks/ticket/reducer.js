/**
 * Internal dependencies
 */
import tickets from './reducers/tickets';
import headerImage, { DEFAULT_STATE as HEADER_IMAGE_DEFAULT_STATE } from './reducers/header-image';
import * as types from './types';
import { PREFIX_TICKETS_STORE } from "@moderntribe/tickets/data/utils";

export const DEFAULT_STATE = {
	headerImage: HEADER_IMAGE_DEFAULT_STATE,
	isSelected: false,
	isSettingsOpen: false,
	isSettingsLoading: false,
	provider: '',
	sharedCapacity: '',
	tempSharedCapacity: '',
	tickets: tickets( undefined, {} ),
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKETS_HEADER_IMAGE:
			return {
				...state,
				headerImage: headerImage( state.headerImage, action ),
			};
		case types.SET_TICKETS_IS_SELECTED:
			return {
				...state,
				isSelected: action.payload.isSelected,
			};
		case types.SET_TICKETS_IS_SETTINGS_OPEN:
			return {
				...state,
				isSettingsOpen: action.payload.isSettingsOpen,
			};
		case types.SET_TICKETS_IS_SETTINGS_LOADING:
			return {
				...state,
				isSettingsLoading: action.payload.isSettingsLoading,
			};
		case types.SET_TICKETS_PROVIDER:
			return {
				...state,
				provider: action.payload.provider,
			};
		case types.SET_TICKETS_SHARED_CAPACITY:
			return {
				...state,
				sharedCapacity: action.payload.sharedCapacity,
			};
		case types.SET_TICKETS_TEMP_SHARED_CAPACITY:
			return {
				...state,
				tempSharedCapacity: action.payload.tempSharedCapacity,
			};
		case types.SET_TICKET_TITLE:
		case types.SET_TICKET_DESCRIPTION:
		case types.SET_TICKET_PRICE:
		case types.SET_TICKET_SKU:
		case types.SET_TICKET_IAC_SETTING:
		case types.SET_TICKET_START_DATE:
		case types.SET_TICKET_START_DATE_INPUT:
		case types.SET_TICKET_START_DATE_MOMENT:
		case types.SET_TICKET_END_DATE:
		case types.SET_TICKET_END_DATE_INPUT:
		case types.SET_TICKET_END_DATE_MOMENT:
		case types.SET_TICKET_START_TIME:
		case types.SET_TICKET_END_TIME:
		case types.SET_TICKET_START_TIME_INPUT:
		case types.SET_TICKET_END_TIME_INPUT:
		case types.SET_TICKET_CAPACITY_TYPE:
		case types.SET_TICKET_CAPACITY:
		case types.SET_TICKET_TEMP_TITLE:
		case types.SET_TICKET_TEMP_DESCRIPTION:
		case types.SET_TICKET_TEMP_PRICE:
		case types.SET_TICKET_TEMP_SKU:
		case types.SET_TICKET_TEMP_IAC_SETTING:
		case types.SET_TICKET_TEMP_START_DATE:
		case types.SET_TICKET_TEMP_START_DATE_INPUT:
		case types.SET_TICKET_TEMP_START_DATE_MOMENT:
		case types.SET_TICKET_TEMP_END_DATE:
		case types.SET_TICKET_TEMP_END_DATE_INPUT:
		case types.SET_TICKET_TEMP_END_DATE_MOMENT:
		case types.SET_TICKET_TEMP_START_TIME:
		case types.SET_TICKET_TEMP_END_TIME:
		case types.SET_TICKET_TEMP_START_TIME_INPUT:
		case types.SET_TICKET_TEMP_END_TIME_INPUT:
		case types.SET_TICKET_TEMP_CAPACITY_TYPE:
		case types.SET_TICKET_TEMP_CAPACITY:
		case types.SET_TICKET_SOLD:
		case types.SET_TICKET_AVAILABLE:
		case types.SET_TICKET_ID:
		case types.SET_TICKET_CURRENCY_SYMBOL:
		case types.SET_TICKET_CURRENCY_POSITION:
		case types.SET_TICKET_PROVIDER:
		case types.SET_TICKET_HAS_ATTENDEE_INFO_FIELDS:
		case types.SET_TICKET_ATTENDEE_INFO_FIELDS:
		case types.SET_TICKET_IS_LOADING:
		case types.SET_TICKET_IS_MODAL_OPEN:
		case types.SET_TICKET_HAS_BEEN_CREATED:
		case types.SET_TICKET_HAS_CHANGES:
		case types.SET_TICKET_HAS_DURATION_ERROR:
		case types.SET_TICKET_IS_SELECTED:
		case types.SET_TICKET_TYPE:
		case types.SET_TICKET_TYPE_DESCRIPTION:
		case types.SET_TICKET_TYPE_ICON_URL:
		case types.SET_TICKET_TYPE_NAME:
		case types.REGISTER_TICKET_BLOCK:
		case types.REMOVE_TICKET_BLOCK:
		case types.REMOVE_TICKET_BLOCKS:
			return {
				...state,
				tickets: tickets( state.tickets, action ),
			};
		case types.SET_UNEDITABLE_TICKETS:
			return {
				...state,
				uneditableTickets: action.payload.uneditableTickets,
				uneditableTicketsLoading: false
			};
		case types.SET_UNEDITABLE_TICKETS_LOADING:
			if (action.loading) {
				return {
					...state,
					uneditableTickets: [],
					uneditableTicketsLoading: true,
				};
			}
			return {
				...state,
				uneditableTicketsLoading: false,
			};
		default:
			return state;
	}
};
