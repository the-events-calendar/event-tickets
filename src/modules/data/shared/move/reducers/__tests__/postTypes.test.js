/**
 * Internal dependencies
 */
import reducer, { DEFAULT_STATE } from '../posts';
import * as types from '../../types';

describe( 'Move postTypes reducer', () => {
	it( 'should show default', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );
	it( 'should fetch types', () => {
		expect(
			reducer( undefined, { type: types.FETCH_POST_TYPES } ),
		).toMatchSnapshot();
	} );
	it( 'should fetch types with success', () => {
		expect(
			reducer( undefined, { type: types.FETCH_POST_TYPES_SUCCESS, data: { posts: { a: 1 } } } ),
		).toMatchSnapshot();
	} );
	it( 'should fetch types with error', () => {
		expect(
			reducer( undefined, { type: types.FETCH_POST_TYPES_ERROR } ),
		).toMatchSnapshot();
	} );
} );
