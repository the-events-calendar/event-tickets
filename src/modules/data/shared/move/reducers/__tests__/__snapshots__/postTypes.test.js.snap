// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move postTypes reducer should fetch types 1`] = `
{
  "isFetching": false,
  "posts": {},
}
`;

exports[`Move postTypes reducer should fetch types with error 1`] = `
{
  "isFetching": false,
  "posts": {},
}
`;

exports[`Move postTypes reducer should fetch types with success 1`] = `
{
  "isFetching": false,
  "posts": {},
}
`;
